src/macros.hpp

```diff
@@ -112,3 +112,32 @@
     namespace Aquamarine {                                                                                                                                                         \
         class name;                                                                                                                                                                \
     }
+
+#define EXTRACT_CLASS_NAME()                                                                                                                                                       \
+    []() -> std::string {                                                                                                                                                          \
```

Comment:
this can be constexpr

Reply:
but can std::string really be constexpr? hitting compiler errors trying todo so

Reply:
std::string_view

Reply:
mhm